Refactor Testing: Unused Variables, Imprecise Asserts

dyastremsky opened the pull request:
A small refactor of some of the testing and models to get rid of unused variables, imprecise asserts, unnecessary imports, and non-recommended syntax.

tanmayv25 commented:
Looks good to me. But @GuanLuo can review mlflow side changes. 

@dyastremsky  Did you use any tool to detect the issues? Can you do something similar for Python backend?

dyastremsky commented:
@tanmayv25 Yup! I saw someone put our repo into LGTM, so I refactored some of the higher-significance alerts. Just added the python_backend repo, it might take some time to analyze.

LGTM link here: https://lgtm.com/projects/g/triton-inference-server/server/

CoderHam commented on qa/common/nightly_email_helper.py:
```diff
@@ -39,7 +39,7 @@
 
 def send(subject: str,
          content: str,
-         attachments=[],
```
Why is this change needed?

dyastremsky replied:
It's because of the attachments.append. Default arguments are instantiated once, not once per call. By using .append, the default argument is changed. So the behavior would vary for every call (i.e. once a new attachment is appended when using the default value, that attachment would already be appended for the next call with the default attachments value).

CoderHam replied:
TIL 😄

CoderHam reviewed (changes requested):
Good work with the cleanup. Is the general convention to not have [] as a default value for params?

CoderHam commented on qa/common/nightly_email_helper.py:
```diff
@@ -39,7 +39,7 @@
 
```
'base64' is not used. You should be able to remove that import

dyastremsky replied:
Removed.

CoderHam commented on qa/common/reporter.py:
```diff
@@ -139,7 +139,8 @@ def post_to_url(url, data):
 
```
Import of 'sys' is not used.

dyastremsky replied:
Removed.

CoderHam commented:
> @dyastremsky Did you use any tool to detect the issues? Can you do something similar for Python backend?

@tanmayv25 Similar tool for python backend
https://lgtm.com/projects/g/triton-inference-server/python_backend?mode=list

We should do a similar run through the client repo

dyastremsky commented:
Thank you both! Addressed the feedback regarding other imports to get rid of. There will likely be more later on, as I only addressed a hundred or so of the higher-priority alerts. Didn't want to do them all at once, as I'm already trying to figure out why [rate_limiter--base is failing](https://gitlab-master.nvidia.com/dl/dgx/tritonserver/-/jobs/29411262), so figured it might make sense to split into a couple PRs.

tanmayv25 commented on qa/L0_rate_limiter/rate_limiter_test.py:
```diff
@@ -267,7 +263,7 @@ def test_single_model_sequence_batching(self):
 
         try:
             model_name = "custom_sequence_int32"
-            self.assertFalse("TRITONSERVER_DELAY_SCHEDULER" in os.environ)
```
Note: this was assertFalse.. We are checking whether TRITONSERVER_DELAY_SCHEDULER was properly unset before running this test.

dyastremsky replied:
Thanks for finding this!
